core/src/main/scala/org/dbpedia/extraction/mappings/InfoboxMappingsExtractor.scala

```diff
@@ -0,0 +1,161 @@
+        t.toWikiText, page.sourceUri, context.ontology.datatypes("xsd:string")))
+
+
+
+    parserFuncQuads ++ templateQuads ++ mappingQuads
+
+  }
+
+  def reduceChildrenToString(propertyNode: PropertyNode) : String = {
+    var answer = ""
+    propertyNode.children.foreach(x => answer += x.toWikiText)
+    return answer
+  }
+  def isNumber(s : String): Boolean = {s.matches("\\d+")}
+
+  def extract_property(str : String, typeOfStr : String) : String = {
+    var answer = ""
+    if ( typeOfStr == "#property") {
+      val pattern = """\{\{#property:([0-9A-Za-z]+)\}\}""".r
+      answer = str match {
+        case pattern(group) => group
+        case _ => ""
+      }
+    } else if (typeOfStr == "#invoke") {
+     val list_words = str.split("""\|""")
+      if (!(list_words(0) == "Wikidata" || list_words(0) == "PropertyLink"))
+        return answer
+      val properties = list_words.filter(s => (s.charAt(0) == 'p' || s.charAt(0) == 'P') && isNumber(s.substring(1)))
+      for ( p <- properties){
+        answer = answer + "/" + p
+      }
+
+      if (answer.size > 0)
+       answer = answer.substring(1)
+
+    }
+    answer
+  }
+  def getPropertyTuples(page : PageNode) : List[(String,String, String)] = {
+    val parserFunctions = ExtractorUtils.collectParserFunctionsFromNode(page)
+
+    val propertyParserFunctions = parserFunctions.filter(p => (p.title.equalsIgnoreCase("#property") &&  //To filter out parser functions with title #property
+      p.children.nonEmpty && // Ignore those that have no children
+      !p.children.head.toString.contains("from") && // Ignore those that have "from" in them for eg {#property:P1308|from=Q824910}
+      p.parent.isInstanceOf[PropertyNode])) // Parent needs to be a PropertyNode to get the key
+
+     (propertyParserFunctions ).map( p =>
+      new Tuple3(p.parent.asInstanceOf[PropertyNode].parent.asInstanceOf[TemplateNode].title.decoded, p.parent.asInstanceOf[PropertyNode].key, extract_property(p.toWikiText, "#property")))
+
+  }
+
+   def getInvokeTuples(page : PageNode) : List[(String, String, String)] = {
+    val parserFunctions = ExtractorUtils.collectParserFunctionsFromNode(page)
+    var invokeFunc = parserFunctions.filter(p => ( p.title.equalsIgnoreCase("#invoke")))
+     invokeFunc = invokeFunc.filter(p => extract_property(p.children.head.toWikiText, "#invoke") != "")
+    (invokeFunc ).map( p => new Tuple3(p.parent.asInstanceOf[PropertyNode].parent.asInstanceOf[TemplateNode].title.decoded, p.parent.asInstanceOf[PropertyNode].key, extract_property(p.children.head.toWikiText, "#invoke")))
+
+  }
+
+  def checkForP856Strings(propertyNode : PropertyNode, lang: Language) : Boolean = {
+
+    for( x <- KeyWordsMapForP856.getOrElse(lang.wikiCode, Map())){
+      if (reduceChildrenToString(propertyNode).contains(x._1))
+        return true
+    }
+
+    return false
+
+  }
+
+  def getP856Tuples(page : PageNode, lang : Language) : List[(String, String, String)] = {
```

Comment:
According to the above comments this could also be generalized to 
`def getDirectTemplateWikidataMappings()`